### PR TITLE
Simplify softmax with cuDNN

### PR DESCRIPTION
**Pull request by okuta**
Please merge cupy/cupy#1736 first.

### CODE REVIEW

**Comment by takagi** [comment]
Would you check CI failure on flake8?

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Would you resolve the conflict?

**Comment by takagi** [comment]
And would you omit 49ba65c from this branch to perform CI without merging cupy/cupy#1734 before?

File: chainer/functions/activation/log_softmax.py [hunk 1]
```diff
@@ -14,10 +13,6 @@
     _algorithm = libcudnn.CUDNN_SOFTMAX_LOG
```
Comment (takagi) [inline chainer/functions/activation/log_softmax.py]: We can remove `libcudnn` module variable.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Would you check the CI failure? We also need to fix calling test for `SoftmaxCrossEntropy`.

**Comment by okuta** [comment]
jenkins, test this please.

**Pull request merged**